Add VIRTIO_NET_F_MRG_RXBUF to virtio-net (#4834)
firecracker-microvm/firecracker: CHANGELOG.md

### Added

- [#4834](https://github.com/firecracker-microvm/firecracker/pull/4834): Add
  `VIRTIO_NET_F_RX_MRGBUF` support to the `virtio-net` device. When this feature

>   is negotiated, guest `virtio-net` driver can perform better memory management

Reviewer: What do you have in mind? The performance improvement will depend highly on the machine and the kernel versions.

>   is negotiated, guest `virtio-net` driver can perform better memory management

Reviewer 2: I mean instead of "better" we could expand what the guest driver will do differently if the feature is negotiated. I didn't mean exact perf numbers.

>   is negotiated, guest `virtio-net` driver can perform better memory management

Reviewer: oh I see. Sorry, I misunderstood the initial comment. Yes that sounds reasonable.

>   is negotiated, guest `virtio-net` driver can perform better memory management

Author: updated message with `more efficient` instead of `better`

> - [#4834](https://github.com/firecracker-microvm/firecracker/pull/4834): Add
>   RX_MRGBUF support to the `virtio-net` device. When this flag is enabled
>   guest network driver can perform better buffer management which in turn
>   improves RX and TX performance.

Reviewer: ```suggestion
- [#4834](https://github.com/firecracker-microvm/firecracker/pull/4834): Add
  support for `VIRTIO_NET_F_MRG_RXBUF` for the `virtio-net` device. When this
  feature is negotiated, guest `virtio-net` driver can perform better memory
  management which in turn improves RX and TX performance.
```

`RX_MRGBUF` is not a valid identifier. Either use, `VIRTIO_NET_F_MRG_RXBUF` or "mergeable buffers". I just saw you used it in the commit message bodies as well. Please fix it there, too.
